### PR TITLE
feat(relocation): Add relocatin provenance

### PR DESCRIPTION
**Pull request by azaslavsky**
This will allow each relocation to track whether it is self-hosted or Saas->Saas. We'll use this information to differentiate logic slightly where appropriate.

### CODE REVIEW

File: src/sentry/models/relocation.py [hunk 1]
```diff
@@ -81,6 +81,15 @@ class EmailKind(Enum):
         def get_choices(cls) -> list[tuple[int, str]]:
             return [(key.value, key.name) for key in cls]
 
+    class Provenance(Enum):
+        SELF_HOSTED = 0
+        SAAS_TO_SAAS = 1
```
Comment (hubertdeng123) [inline src/sentry/models/relocation.py:86]: nit: technically shouldn't this be just SAAS?

**Review by hubertdeng123 — APPROVED** [review]

File: src/sentry/migrations/0733_relocation_provenance.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ]
+
+    operations = (
+        [
+            migrations.SeparateDatabaseAndState(
+                database_operations=(
+                    [
+                        migrations.RunSQL(
+                            """
+                        ALTER TABLE "sentry_relocation" ADD COLUMN "provenance" integer NOT NULL DEFAULT 1;
```
Comment (markstory) [inline src/sentry/migrations/0733_relocation_provenance.py:34]:
```suggestion
                        ALTER TABLE "sentry_relocation" ADD COLUMN "provenance" integer NOT NULL DEFAULT 0;
```

Default values should be aligned.

File: src/sentry/models/relocation.py [hunk 3]
```diff
@@ -81,6 +81,15 @@ class EmailKind(Enum):
         def get_choices(cls) -> list[tuple[int, str]]:
             return [(key.value, key.name) for key in cls]
 
+    class Provenance(Enum):
```
Comment (markstory) [inline src/sentry/models/relocation.py:84]: If you inherit from IntEnum you won't have to use `.value` when using the enum.

File: src/sentry/migrations/0733_relocation_provenance.py (hunk 2 above)
Comment (azaslavsky) [reply src/sentry/migrations/0733_relocation_provenance.py]: Done.

File: src/sentry/models/relocation.py (hunk 3 above)
Comment (azaslavsky) [reply src/sentry/models/relocation.py]: Done.

File: src/sentry/models/relocation.py (hunk 1 above)
Comment (azaslavsky) [reply src/sentry/models/relocation.py:86]: Think `SAAS_TO_SAAS` is a bit clearer and more greppable in this context.

**Review by markstory — APPROVED** [review]

**Pull request merged**